packages/api-contract/src/base/Blueprint.ts

```diff
@@ -52,11 +52,12 @@ export class Blueprint<ApiType extends ApiTypes> extends Base<ApiType> {
     return this.#tx;
   }
 
-  #deploy = (constructorOrId: AbiConstructor | string | number, { gasLimit = BN_ZERO, salt, value = BN_ZERO }: BlueprintOptions, params: unknown[]): SubmittableExtrinsic<ApiType, BlueprintSubmittableResult<ApiType>> => {
+  #deploy = (constructorOrId: AbiConstructor | string | number, { gasLimit = BN_ZERO, storageDepositLimit = BN_ZERO, salt, value = BN_ZERO }: BlueprintOptions, params: unknown[]): SubmittableExtrinsic<ApiType, BlueprintSubmittableResult<ApiType>> => {
     return this.api.tx.contracts
       .instantiate(
         value,
         gasLimit,
+        storageDepositLimit,
```

Comment:
Backwards compat is always a HUGE issue. So, in this case (luckily) the number of paramaters changed, so it would be a case of -

```js
this.api.tx.contracts.instantiate.meta.arags.length === 6
  ? this.api.tx.contracts.instantiate(value, gasLimit, storageDepositLimit, codeHash, params, salt)
  : this.api.tx.contracts.instantiate(value, gasLimit, codeHash, params, salt)
```

Reply:
is this change necessary in case api-contract version is different than api version? 
it would mean overloading the generated interfaces somehow.

Reply:
IT is needed since it depends on the chain it is connected to - so it needs to be detected via on-chain metadata. You can have the most recent API with an ancient node.

Reply:
i see... tried a few things yesterday but unable to figure out how to do that. is there a PR where you did something similar that you can point me to?

Reply:
So, it is a bit of a nightmare, since these things are short-lived - so I generally allow 3 months leeway before ripping it out, so the examples that were here (contracts specific) have been removed.

Will dig around the older versions and find it. (There is a similar approach in the apps UI as well, which tries to also give around 3 months.... but generally they end up living a "bit" longer)

Reply:
Here is an example for the apps UI that does the same thing -

https://github.com/polkadot-js/apps/blob/master/packages/page-democracy/src/Overview/Fasttrack.tsx#L59-L63

(There are millions of them in there... sadly, mostly in the API it is always only contracts since there are convenience wrappers around... will find an old release having some)

Reply:
got it now! you have to `ts-ignore` the old style

Reply:
Yes, it is slightly messy code-wise :(

… but better than dealing with the “it doesn’t work-anymore” flood of support…

Reply:
I think i got everything but still can't call. I get this   rpc core error
https://github.com/polkadot-js/apps/pull/6657#issuecomment-991983593

There is also this contract query, but not sure if adding an extra param there would solve it and